components/BottomNavigation/src/MDCBottomNavigationView.m

```diff
@@ -0,0 +1,258 @@
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCBottomNavigationView.h"
+
+#import "MaterialMath.h"
+#import "MaterialRTL.h"
+#import "private/MDCBottomNavigationCell.h"
+
+static NSString *const kMDCBottomNavigationViewBadgeColorString = @"badgeColor";
+static NSString *const kMDCBottomNavigationViewBadgeValueString = @"badgeValue";
+static NSString *const kMDCBottomNavigationViewImageString = @"image";
+static NSString *const kMDCBottomNavigationViewTitleString = @"title";
+static NSString *const kMDCBottomNavigationViewNewString = @"new";
+
+@interface MDCBottomNavigationView ()
+
+@property(nonatomic, strong) NSMutableArray<MDCBottomNavigationCell *> *navBarCells;
+@property(nonatomic, assign) UIUserInterfaceLayoutDirection layoutDirection;
+@property(nonatomic, strong) UIView *navBarContainerView;
+
+@end
+
+@implementation MDCBottomNavigationView
+
+- (instancetype)initWithFrame:(CGRect)frame {
+  self = [super initWithFrame:frame];
+  if (self) {
+    [self commonMDCBottomNavigationViewInit];
+  }
+  return self;
+}
+
+- (instancetype)initWithCoder:(NSCoder *)aDecoder {
+  self = [super initWithCoder:aDecoder];
+  if (self) {
+    [self commonMDCBottomNavigationViewInit];
+  }
+  return self;
+}
+
+- (void)commonMDCBottomNavigationViewInit {
+  self.backgroundColor = [UIColor whiteColor];
+  self.autoresizingMask = (UIViewAutoresizingFlexibleTopMargin | UIViewAutoresizingFlexibleWidth);
+  
+  _selectedColor = [UIColor blackColor];
+  _unselectedColor = [UIColor grayColor];
+  _layoutDirection = self.mdc_effectiveUserInterfaceLayoutDirection;
+
+  // Content in bottom navigation always uses the width of the device portrait orientation width.
+  CGSize appSize = [[UIScreen mainScreen] applicationFrame].size;
+  CGFloat minDimension = MIN(appSize.width, appSize.height);
+  CGRect adjustedFrame = CGRectMake(0, 0, minDimension, self.frame.size.height);
+  _navBarContainerView = [[UIView alloc] initWithFrame:adjustedFrame];
+  _navBarContainerView.autoresizingMask = (UIViewAutoresizingFlexibleHeight |
+                                           UIViewAutoresizingFlexibleLeftMargin |
+                                           UIViewAutoresizingFlexibleRightMargin );
+  [self addSubview:_navBarContainerView];
+  _navBarCells = [NSMutableArray array];
+}
+
+- (void)layoutSubviews {
+  [super layoutSubviews];
+
+  [self layoutCellsWithLayoutDirection:self.layoutDirection];
+  self.navBarContainerView.center = CGPointMake(CGRectGetMidX(self.bounds),
+                                                CGRectGetMidY(self.bounds));
+}
+
+- (void)layoutCellsWithLayoutDirection:(UIUserInterfaceLayoutDirection)layoutDirection {
+  NSInteger numItems = self.navBarItems.count;
+  NSInteger i = 0;
+  CGSize navBarSize = self.navBarContainerView.bounds.size;
+  CGFloat itemWidth = navBarSize.width / numItems;
+  if (layoutDirection == UIUserInterfaceLayoutDirectionRightToLeft) {
```

Comment:
Most of our client developers will be going through this code in LeftToRight mode, so handle that case first.

Alternatively, consider laying out everything in standard LTR mode and mirror cells if required.

if (self.mdf_effectiveInterface == RTL) {
  for (cell in self.navBarCells) {
    cell.frame = MDFRectMirror(cell.frame, CGRectGetWidth(self.navBarContainerView.frame));
  }
}